bridges/otelzap/encoder.go

```diff
@@ -55,15 +55,13 @@ func (m *objectEncoder) calculate(o *namespace) {
 }
 
 func (m *objectEncoder) AddArray(key string, v zapcore.ArrayMarshaler) error {
-	// TODO: Use arrayEncoder from a pool.
 	arr := &arrayEncoder{}
```

Comment:
Wouldn't pre-allocating the slice in `arrayEncoder` reduce the amount of heap allocations (similarly to `newObjectEncoder(2)` for object encoder)?

This can be handled in a separate PR.